src/main/kotlin/Menu.kt

```diff
@@ -0,0 +1,51 @@
+class Menu<T>(private val title: String, private val items: MutableList<T>, private val onItemSelected: (T) -> Unit, private val onCreateNewItem: () -> T) {
+
+    fun display() {
+        while (true) {
+            println("\n$title")
+            println("0. Создать новый")
+
+            items.forEachIndexed { index, item ->
+                val itemName = if (item is Archive) item.name else (item as? Note)?.title ?: "Элемент"
```

Comment:
```suggestion
                val itemName = if (item is Archive) {
                     item.name
                } else { 
                     (item as? Note)?.title
                 } ?: "Элемент"
```

А еще лучше заменить всю конструкцию на when